### DISCUSSION TITLE
Not recovering from internet outage

### DISCUSSION THREAD

**Discussion by tdcockers** [abraunegg/onedrive#2490] [Q&A]
I've had this on a couple of systems now, they are different sites but generally configured the same (Onedrive syncing Sharepoint to Nethserver, hosted on xcp-ng, Vyos doing routing), so not ruling out an environmental issue. 

The issue arises (sometimes) when the internet fails, but the server that Onedrive is on keeps running. journalctl logs show repeating entries along the lines of "no response from server, internet/connectivity issue?", but these errors keep going once the internet service is restored. Restarting the onedrive service is all that is needed to get it going again.

Is this a bug or likely related to environment? Is there a way of setting a watchdog on the services so that it attempts to periodically restart when it is stuck in this loop?

**Comment by abraunegg** [comment]
@tdcockers 

> The issue arises (sometimes) when the internet fails, but the server that Onedrive is on keeps running. journalctl logs show repeating entries along the lines of "no response from server, internet/connectivity issue?", but these errors keep going once the internet service is restored.

So what 'should' be happening here is, once the Internet is reachable again, the process identifies this and carries on.

> Is this a bug or likely related to environment?

Too hard to say accuratly - but potentially is a 'bug'

The issue here is - I test the following explicitly:

* Block HTTPS at my firewall
  * The result is the client detects that OneDrive service is unavailable, goes into the applicable loop, comes out of loop when HTTPS is unblocked
* Block DNS
  * The result is that applicable resolution errors occur, goes into the applicable loop, comes out of loop when DNS is unblocked
* Unplug relevant upstream network cable causing an 'outage' upstream
  * The result is the client detects that OneDrive service is unavailable, goes into the applicable loop, comes out of loop when connectivity is returned

So .. these 'tests' that work tells me that the application is 'working' correctly for when the Internet access to OneDrive *disappears* ....

Can you come up with a 100000% reproducible test case here - so that I can reproduce the issue you are seeing. If I can reproduce, I can look at how to fix it.

**Comment by tdcockers** [comment]
Yes I'll do some further testing when I can... was just putting out feelers to see if the same issue had occurred elsewhere. I haven't been able to force it to reproduce either, in one case I noticed this morning, one of the jobs was stuck in a loop but the other 3 to the same tenant were all fine. Will keep you posted.
